Inc/GraphicsMemory.h

```diff
@@ -214,7 +214,20 @@ namespace DirectX
             // Private implementation.
             class Impl;
 
-            GraphicsResource __cdecl AllocateImpl(size_t size, size_t alignment);
+            GraphicsResource __cdecl AllocateImpl(size_t size, size_t alignment){ //Allocate memory using the global new operatr
```

Comment:
For the Impl pattern, this implementation should be in `GraphicsMemory.cpp` and not inline in the header.

How do you see a failed allocation happening here that is not resulting in a ``bad_malloc`` throw?

```
GraphicsMemory::AllocateImpl
    GraphicsMemory::Impl::Allocate
        DeviceAllocator::Alloc
```

``DeviceAllocator::Alloc`` explicitly throw a ``std::bad_alloc`` if it can't get a page. It will throw a `std::runtime_error` exception if `LinearAllocatorPage::Suballocate` fails although this scenario is not likely to happen in a working program.

Calling 'new' here is also not the right implementation since it needs to use the LinearAllocator.